src/test/java/com/artipie/management/FakeUsers.java

```diff
@@ -35,27 +42,94 @@
  *  implementation, enable DeleteUserSliceTest#returnsNotFoundIfCredentialsAreEmpty() test method.
  *  `Users.FromStorageYaml` should stay in artipie and be removed from this repository.
  */
-@SuppressWarnings("PMD.AvoidDuplicateLiterals")
 public final class FakeUsers implements Users {
 
+    /**
+     * Users.
+     */
+    private final Map<User, Password> users;
+
+    /**
+     * Primary ctor.
+     * @param users Users and passwords
+     */
+    public FakeUsers(final Map<User, Password> users) {
+        this.users = users;
+    }
+
+    /**
+     * Empty users.
+     */
+    public FakeUsers() {
+        this(Collections.emptyMap());
+    }
+
+    /**
+     * Primary ctor.
+     * @param name User name
+     */
+    public FakeUsers(final String name) {
+        this(new HashMap<>(new MapOf<>(new MapEntry<>(new User(name), new Password()))));
+    }
+
     @Override
     public CompletionStage<List<User>> list() {
-        throw new NotImplementedException("Not implemented yet");
+        final CompletableFuture<List<User>> res = new CompletableFuture<>();
+        res.complete(new ArrayList<>(this.users.keySet()));
+        return res;
     }
 
     @Override
     public CompletionStage<Void> add(final User user, final String pswd,
         final PasswordFormat format) {
-        throw new NotImplementedException("Not implemented yet");
+        this.users.put(user, new Password(pswd, format));
+        return CompletableFuture.allOf();
     }
 
     @Override
     public CompletionStage<Void> remove(final String username) {
-        throw new NotImplementedException("Not implemented yet");
+        this.users.keySet().stream().filter(user -> user.name().equals(username))
+            .findFirst().ifPresent(this.users::remove);
```

Comment:
@olenagerasimova `.findFirst().ifPresent(...)` could be simplified to `.forEach(...)`

Reply:
@olegmoz hm, I do not see how using `forEach` is a simplification as inside it I should add if-statement to check user name...